osu.Game/Tests/CleanRunHeadlessGameHost.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Platform;
+
+namespace osu.Game.Tests
+{
+    /// <summary>
+    /// A headless host which cleans up before running (removing any remnants from a previous execution).
+    /// </summary>
+    public class CleanRunHeadlessGameHost : HeadlessGameHost
+    {
+        public CleanRunHeadlessGameHost(string gameName = @"", bool bindIPC = false, bool realtime = true)
+            : base(gameName, bindIPC, realtime)
+        {
+            Storage.DeleteDirectory(string.Empty);

```

Comment:
This is not going to work for OsuTestCase, since that has a different Guid with every run.

Reply:
We should probably be deleting _after_ every run, similar to how the base `TestCase` does it.

Reply:
the guid is passed in as gameName..

Reply:
Yes? That's what's going to stop it from working.

Reply:
<img width="685" alt="screen shot 2018-02-08 at 5 27 07 pm" src="https://user-images.githubusercontent.com/1329837/35962582-4f220ed0-0cf5-11e8-9d1c-a055271539d1.png">

![screen shot 2018-02-08 at 5 27 18 pm](https://user-images.githubusercontent.com/1329837/35962590-539fe4aa-0cf5-11e8-9fb1-794d45ce3903.png)

Reply:
this is only to ensure that we have a clean environment, not to guarantee cleanup.

we can experiment with adding it to disposal logic, but i found that to be unreliable.

Reply:
Right...